packages/world/src/systems/MineSystem.sol

```diff
@@ -158,7 +159,11 @@ library MineLib {
 contract MineSystem is System {
   using ObjectTypeLib for ObjectTypeId;
 
-  function _removeBlock(EntityId entityId, Vec3 coord) internal {
+  function _removeBlock(EntityId entityId, ObjectTypeId objectTypeId, Vec3 coord) internal {
+    if (objectTypeId.isSeed()) {
```

Comment:
What happens if you call mine on a seed that is fully grown? You'd get the seed right. That feels odd cuz the seed has already grown

Feel like we should either revert the mine, and player has to call growSeed, or we should call growSeed here then run mine

Reply:
hmm but shouldn't the client figure that out? could even do a batch call, that way we keep mining cheaper and simpler

Reply:
our client will deff do it (but many ppl will run scripts), and so I mean the way it is right now, technically the physics allows you to store energy in a seed and then put it back into the local energy pool, by mining seeds instead of calling grow. that feels weird and shouldnt be allowed

Reply:
how would this work in the case of trees? should it mine the base of the trunk or something like that? maybe we can just prevent it for now

Reply:
yea simplest might be to just revert the mine tx

Reply:
I guess this also applies for seeds that are on top of blocks being mined

Reply:
yee just revert. Our client will never get into this state anyways so user wont know about it